frontend/src/components/professor/GameManagement/RanksAndBadges/BadgeCreationForm.js

```diff
@@ -0,0 +1,137 @@
+
+function BadgeCreationForm(props) {
+  const [isFetching, startAddition] = useTransition()
+  const [errorMessage, setErrorMessage] = useState(undefined)
+
+  // Formik does not support field type input, so we have to do it manually
+  const [chosenFile, setChosenFile] = useState(null)
+
+  const checkboxRef = useRef()
+
+  const addNewBadge = (values) => {
+    UserService.addBadge(
+      values.title,
+      values.description,
+      chosenFile,
+      values.customValue,
+      checkboxRef.current?.checked.toString(),
+      values.type
+    )
+      .then(() => {
+        props.setModalOpen(false)
+        successToast('Nowa odznaka została dodana pomyślnie')
+        props.onSuccess()
+      })
+      .catch((error) => {
+        setErrorMessage(error.response?.data?.message ?? ERROR_OCCURRED)
+      })
+  }
+
+  return (
+    <Formik
+      initialValues={{
+        title: '',
+        description: '',
+        customValue: '',
+        type: Object.keys(BadgeType)[0]
+      }}
+      validate={(values) => {
+        const errors = {}
+        if (!values.title) errors.title = FIELD_REQUIRED
+        if (!values.description) errors.description = FIELD_REQUIRED
+        if (!values.customValue) errors.customValue = FIELD_REQUIRED
+        return errors
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        if (!chosenFile) {
+          setErrorMessage(FILE_INPUT_REQUIRED)
+        } else {
+          startAddition(() => {
+            addNewBadge(values)
+            setSubmitting(false)
+          })
+        }
+      }}
+    >
+      {({ handleSubmit, values }) => (
+        <Form onSubmit={handleSubmit}>
+          <Container>
+            <Row className='mx-auto'>
+              {FormCol('Nazwa odznaki', 'text', 'title', 12, {
+                errorColor: props.theme.danger
+              })}
+              {FormCol('Opis', 'textarea', 'description', 12, {
+                errorColor: props.theme.danger
+              })}
+              {FormCol('Wartość odblokowująca', 'number', 'customValue', 12, {
+                errorColor: props.theme.danger
+              })}
+              {FormCol('Typ odznaki', 'select', 'type', 12, {
+                errorColor: props.theme.danger,
+                options: Object.keys(BadgeType).map((key) => ({ name: key, value: key }))
+              })}
+              <Col md={12}>
+                <h6>Ikona dla odznaki</h6>
+                <input
+                  name='file'
+                  type='file'
+                  onChange={(event) => {
+                    setChosenFile(event.currentTarget.files[0])
+                  }}
+                />
+              </Col>
+              {values.type === BadgeType.TOP_SCORE ? (
+                <Col md={12}>
+                  <Form.Check
+                    ref={checkboxRef}
+                    className={'my-3'}
+                    label={'Zastosuj tylko dla grupy, do której należy student.'}
```

Comment:
Nie rozumiem tej opcji, zwłaszcza że jest u prowadzącego - chodzi o to że tylko grupy ze studentami przeliczają tą odznakę?

Reply:
Chodzi o to, że dla odznak typu TOP_SCORE można dodać odznakę, która liczy ten top score z wszystkich studentów w systemie lub tylko z grupy do której należy student X